sdk/avalon_sdk/connector/blockchains/ethereum/ethereum_wrapper.py

```diff
@@ -207,11 +209,49 @@ def execute_unsigned_transaction(self, tx_dict):
         Transaction receipt on success or None on error.
         """
         tx_hash = self.__w3.eth.sendTransaction(tx_dict)
-        tx_receipt = self.__w3.eth.waitForTransactionReceipt(tx_hash)
+        tx_receipt = self.__w3.eth.waitForTransactionReceipt(
+            tx_hash, timeout=10, poll_latency=0.2)
         logging.debug("Executed transaction hash: %s, receipt: %s",
                       format(tx_hash.hex()), format(tx_receipt))
         return tx_receipt
 
+    def build_exec_txn(self, contract_func):
+        """
+        Build transaction parameters and execute transaction. This function
+        makes an attempt to commit transactions 3 times if it fails on the
+        first attempt. Between each attempt, it sleeps for a random time
+        period less than 5 seconds.
+
+        Parameters:
+        contract_func   Populated function instance to be invoked
+        Returns:
+        txn_receipt     Transaction receipt, if the transaction is successful
+        """
+        retry_count = 0
+        while True:
+            try:
+                txn_dict = contract_func.buildTransaction(
+                    self.get_transaction_params())
+                return self.execute_transaction(txn_dict)
+            except Exception as ex:
+                # There is a possibility of timeout when a transaction is not
+                # mined. One of the reasons to not get mined could be that 2
+                # transactions had the same nonce (transaction sequence no.).
+                # So, a retry is worth doing at this point with a new nonce.
+                retry_count += 1
+                # Sleep for random number of seconds less than 5
+                time.sleep(random.randint(0, 5))
+                logging.warn("Exception in writing transaction to"
+                             " blockchain: {}".format(ex))
+                # Retry 3 times only
+                if retry_count == 3:
+                    if isinstance(ex, web3.exceptions.TimeExhausted):
+                        raise TimeoutError(
+                            "Transactions are being submitted at a faster "
+                            "rate than they can be processed.")
+                    else:
+                        raise
```

Comment:
Is raise Exception type missing..

Reply:
No `raise` is like a pass through when used alone. It will forward the exception received.
Just for understanding for all. :)
```
>>> try:
...     raise TimeoutError("Timeout")
... except Exception as ex:
...     raise
...
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
TimeoutError: Timeout
>>> try:
...     raise TimeoutError("Timeout")
... except Exception as ex:
...     raise Exception("Some exception occurred")
...
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
**TimeoutError: Timeout**

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
**Exception: Some exception occurred**
>>>
```